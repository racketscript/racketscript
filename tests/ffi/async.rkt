#lang racketscript/base
(require "promise-handlers.rkt"
         racketscript/interop)

;; see also promises.rkt

;; ## prints (everything after err is aborted)
;; Initial
;; handler1:
;;   do that (from catch)

;; using async/await
($/define/async (go)
  (define result ($/await (mkpromise)))
  (define res2 ($/await (handler1 result)))
  (define res3 ($/await (handler2 result))) ;; skipped
  (#js*.console.log #js"final result:") ;; skipped
  (#js*.console.log res3))

($> (go)
    (catch errhandle))
  
