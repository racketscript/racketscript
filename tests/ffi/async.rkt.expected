Initial
handler 1
  do that
